### PR TITLE
Maharjun | BAH-511 | Making `Go To IPD Dashboard` button as configurable

### PR DESCRIPTION
**Pull request by maharjunm**
https://bahmni.atlassian.net/browse/BAH-511

### CODE REVIEW

**Comment by sumanmaity112** [comment]
Hi @maharjunm,
Please fix the **codeclimate** and **travis** issues

**Comment by maharjunm** [comment]
@sumanmaity112  I made changes according to feedback. Please go through commit and let me know if you need more changes

**Comment by sumanmaity112** [comment]
Hi @maharjunm, 
Looks ok. Can you please move the JIRA card to `Needs code review` lane.

**Comment by sai789** [comment]
@sumanmaity112 and @maharjunm 
We moved this card to Code review lane

**Comment by SwathiVarkala** [comment]
Please do single data binding where ever applicable to reduce no of angular watches degrading performance.

File: ui/app/clinical/consultation/controllers/consultationController.js [hunk 1]
```diff
@@ -147,6 +151,7 @@ angular.module('bahmni.clinical').controller('ConsultationController',
                 var appExtensions = clinicalAppConfigService.getAllConsultationBoards();
                 $scope.availableBoards = $scope.availableBoards.concat(appExtensions);
                 $scope.showSaveConfirmDialogConfig = appService.getAppDescriptor().getConfigValue('showSaveConfirmDialog');
+                $scope.ipdButtonConfig = appService.getAppDescriptor().getConfigValue('ipdButton') || { "hideGoToIPDButton": true };
```
[the remark this answers is not in the document]
Comment (pramidatumma) [reply ui/app/clinical/consultation/controllers/consultationController.js]: We have discussed these comments and details in the PR with Swathi and Bindu.
We will make below code changes:
1. Remove hide functionality and keep the existing functionality of showing the ipd button based on privilege
2. Rename "ipdButton" and "ipdaButtonConfig" to "adtNavigationConfig". 
3. The attributes of this config are forwardUrl, privilege, title

File: ui/app/common/patient-search/controllers/patientsListController.js [hunk 2]
```diff
@@ -116,6 +116,7 @@ angular.module('bahmni.common.patientSearch')
 
         $scope.forwardPatient = function (patient) {
             var options = $.extend({}, $stateParams);
+            $rootScope.patientAdmitLocationStatus = patient.Status;
```
[the remark this answers is not in the document]
Comment (pramidatumma) [reply ui/app/common/patient-search/controllers/patientsListController.js]: Checked that it's not being used. We will remove it

File: ui/app/clinical/dashboard/views/clinicalDashboardHeader.html [hunk 3]
```diff
@@ -5,8 +5,8 @@
         <div class="opd-header-bottom tabs">
             <bm-back-links></bm-back-links>
             <recent-patients ng-if="enablePatientSearch()"></recent-patients>
-            <a show-if-privilege="app:adt" ng-click= "displayConfirmationDialog($event)" class="back-btn"  ng-href="../adt/#/patient/{{patient.uuid}}/visit/{{consultation.visitUuid}}/"
-               title="Go to IPD dashboard">
+            <a show-if-privilege="{{ipdButtonConfig.privilege}}" ng-if="!isGoToIPDButtonHidden()" ng-click= "displayConfirmationDialog($event)" class="back-btn" ng-href="{{generateBedManagementURL(consultation.uuid)}}"
```
[the remark this answers is not in the document]
Comment (pramidatumma) [reply ui/app/clinical/dashboard/views/clinicalDashboardHeader.html]: 1. It is taken care of
2. We will rename to "adtNavigationURL"

File: ui/app/clinical/constants.js [hunk 4]
```diff
@@ -67,8 +67,9 @@ Bahmni.Clinical.Constants = (function () {
         mandatoryVisitConfigUrl: "config/visitMandatoryTab.json",
         defaultExtensionName: "default",
         bacteriologyConstants: bacteriologyConstants,
-        globalPropertyToFetchActivePatients: 'emrapi.sqlSearch.activePatients'
-
+        globalPropertyToFetchActivePatients: 'emrapi.sqlSearch.activePatients',
+        adtPrivilege: "app:adt",
+        forwardUrl: "../adt/#/patient/{{patientUuid}}/visit/{{visitUuid}}/"
```
Comment (SwathiVarkala) [inline ui/app/clinical/constants.js]: forwardUrl -> adtForwardUrl
Naming should be improved

File: ui/test/unit/clinical/controllers/consultationController.spec.js [hunk 5]
```diff
@@ -302,20 +312,24 @@ describe("ConsultationController", function () {
...
-            createController();
             appDescriptor = {
                 formatUrl: function (url) {
                     return url;
                 },
-                getConfigValue: function () {
-                    return true;
+                getConfigValue: function (value) {
+                    console.log(value);
+                    if(value !== 'ipdButton') {
```
Comment (SwathiVarkala) [inline ui/test/unit/clinical/controllers/consultationController.spec.js]: ipdButton?

File: ui/test/unit/clinical/controllers/consultationController.spec.js [hunk 6]
```diff
@@ -256,6 +254,18 @@ describe("ConsultationController", function () {
             rootScope = $rootScope;
             controller = $controller;
         });
+        appDescriptor = {
+            formatUrl: function (url) {
+                return url;
+            },
+            getConfigValue: function (value) {
+                if(value !== 'ipdButton') {
```
Comment (SwathiVarkala) [inline ui/test/unit/clinical/controllers/consultationController.spec.js]: ipdButton?
ipdButton is renamed to adtNavigationConfig right?
If that is the case then how are tests passing with still test depending on ipdButton?

**Comment by SwathiVarkala** [comment]
Looks good to me. We can go ahead on this.

**Pull request merged**